ereolen_support/ereolen_support.helper.inc

```diff
@@ -0,0 +1,255 @@
+<?php
+
+/**
+ * @file
+ * Contains helper functions for the support form.
+ */
+
+/**
+ * Search for a drupal user email in Jira.
+ *
+ * @param \stdClass $user
+ *   A drupal user.
+ *
+ * @return bool|mixed
+ *   A jira user or FALSE if no user was found.
+ *
+ * @throws \Exception
+ */
+function _ereolen_support_search_jira_user($user, $form_state) {
+  $form_email = $form_state['values']['ereolen_support_email'];
+  $requestParts = _ereolen_support_get_request_parts('GET');
+  $result = drupal_http_request($requestParts['jira_path'] . '/user/search?includeInactive=true&username=' . $form_email, $requestParts['options']);
+  $resultArray = json_decode($result->data);
+  _ereolen_support_log_error($result);
+
+  // Searches for matches in full name, username and email, so we may have
+  // multiple results.
+  if(!empty($resultArray)) {
+    foreach ($resultArray as $jiraUser) {
+      if($jiraUser->name == $form_email) {
```

Comment:
Maybe check that `$jiraUser` is an object before trying to access properties on stuff fetched from a remote server. If it is not an object og don't have `name` this would throw a fatal error. As the json given from the remote server have not been validated.